### PR TITLE
Introduce ToguruClientGenerator

### PR DESCRIPTION
**Pull request by mriceron**
The scope of the PR is to `ToguruClientGenerator` in order to make it possible to create a Toguru client that does not use the default fetcher for toggling data and make possible to test more complex scenarios.

```
const stubToguruClientByActivationContext = toguruClientGenerator({
  config: { endpoint: '', refreshIntervalMs: 10000 },
  fetcher: () =>
    Promise.resolve({
      toggles: [
        {
          id: 'some-id',
          tags: {},
          activations: [
            {
              attributes: {
                someAttribute: ['some-value'],
              },
              rollout: {
                percentage: 100,
              },
            },
          ],
        },
      ],
      sequenceNo: 0,
    }),
})
```

### CODE REVIEW

File: src/express/bridge.ts [hunk 1]
```diff
@@ -71,3 +62,12 @@ export const expressClient = (config: ExpressConfig) => (req: Request): Toggling
         togglesForService: (service: string) => config.client(context).togglesForService(service),
     }
 }
+
```
Comment (omakoleg) [inline src/express/bridge.ts:65]: why this was moved ?

File: src/express/testHelpers.ts [hunk 2]
```diff
@@ -4,12 +4,12 @@ import { Toggle } from '../models/Toggle'
 import { Toggles } from '../models/Toggles'
 import { TogglingApi } from '../client'
 
+export const stubClient = (overrides: ToggleState[]) => (_: Request): TogglingApi => ({
```
Comment (omakoleg) [inline src/express/testHelpers.ts:7]: here as well I dont see changes

**Review by omakoleg — COMMENTED** [review]
LGTM.

Can you also please mention that sample from PR description in docs on how to use it to achieve this special activations in tests ?

File: src/express/bridge.ts (hunk 1 above)
Comment (mriceron) [reply src/express/bridge.ts:65]: The variable was used before the declaration

File: src/express/testHelpers.ts (hunk 2 above)
Comment (mriceron) [reply src/express/testHelpers.ts:7]: Same as above just fixing limiting issues

**Comment by mriceron** [comment]
> LGTM.
> 
> 
> 
> Can you also please mention that sample from PR description in docs on how to use it to achieve this special activations in tests ?

Yeah sure I'm gonna extend the Readme 👍

File: src/client.ts [hunk 3]
```diff
@@ -33,13 +33,21 @@ export type TogglingApi = {
...
 
     const refreshToguruData = () =>
-        fetchToguruData(endpoint)
-            .then((td) => (toguruData = td))
+        fetcher(endpoint)
+            .then((td) => {
+                toguruData = td
+                console.info(`Refreshed toguru data, seqNo: ${toguruData.sequenceNo}`)
+            })
             .catch((e) => console.warn(`Unable to refresh toguru data: ${e}`))
```
Comment (kiranbayram) [inline src/client.ts:51]: shouldn't this be error level ?

**Comment by mriceron** [comment]
Merging to the branch for the 2.2.0 release

**Pull request merged**